video/agon_audio.h

```diff
@@ -76,17 +78,20 @@ void setSampleRate(uint16_t sampleRate) {
 	// detach the old sound generator
 	if (soundGenerator) {
 		soundGenerator->play(false);
-		for (auto channelPair : audioChannels) {
-			auto channel = channelPair.second;
-			channel->detachSoundGenerator();
+		for (const auto &channelPair : audioChannels) {

```

Comment:
there were two issues here that I'd observed when trying to reset the underlying sample rate following playing a mod file

firstly, looping the iterator was turning up a duff channel - hence the inclusion of the belt & braces check on the following line

secondly the iterator was inefficient since it wasn't actually using a reference to the channelPair